src/code/z_player_lib.c

```diff
@@ -1,153 +1,1021 @@
+        }
+        Player_SetModelGroup(player, Player_ActionToModelGroup(player, player->itemActionParam));
+        func_80123140(globalCtx, player);
+        if (player->unk_B62 != 0) {
+            player->unk_B62 = 1;
+        }
+    }
+}
+
+void func_80123D50(GlobalContext* globalCtx, Player* player, s32 item, s32 actionParam) {
+    func_80114FD0(globalCtx, item, player->heldItemButton);
+
+    if (item != ITEM_BOTTLE) {
+        player->heldItemId = item;
+        player->itemActionParam = actionParam;
+    }
+
+    player->heldItemActionParam = actionParam;
+}
+
+void func_80123DA4(Player* player) {
+    player->unk_730 = NULL;
+    player->stateFlags2 &= ~PLAYER_STATE2_2000;
+}
+
+void func_80123DC0(Player* player) {
+    if ((player->actor.bgCheckFlags & 1) ||
+        (player->stateFlags1 & (PLAYER_STATE1_8000000 | PLAYER_STATE1_800000 | PLAYER_STATE1_200000)) ||
+        (!(player->stateFlags1 & (PLAYER_STATE1_80000 | PLAYER_STATE1_40000)) &&
+         ((player->actor.world.pos.y - player->actor.floorHeight) < 100.0f))) {
+        player->stateFlags1 &= ~(PLAYER_STATE1_40000000 | PLAYER_STATE1_80000 | PLAYER_STATE1_40000 |
+                                 PLAYER_STATE1_20000 | PLAYER_STATE1_10000 | PLAYER_STATE1_8000);
+    } else if (!(player->stateFlags1 & (PLAYER_STATE1_200000 | PLAYER_STATE1_80000 | PLAYER_STATE1_40000))) {
+        player->stateFlags1 |= PLAYER_STATE1_80000;
+    } else if ((player->stateFlags1 & PLAYER_STATE1_40000) && (player->transformation == PLAYER_FORM_DEKU)) {
+        player->stateFlags1 &=
+            ~(PLAYER_STATE1_40000000 | PLAYER_STATE1_20000 | PLAYER_STATE1_10000 | PLAYER_STATE1_8000);
+    }
+    func_80123DA4(player);
+}
+
+void func_80123E90(GlobalContext* globalCtx, Actor* actor) {
+    Player* player = GET_PLAYER(globalCtx);
+
+    func_80123DC0(player);
+    player->unk_730 = actor;
+    player->unk_A78 = actor;
+    player->stateFlags1 |= PLAYER_STATE1_10000;
+    func_800DFD78(Play_GetCamera(globalCtx, CAM_ID_MAIN), 8, actor);
+    Camera_ChangeMode(Play_GetCamera(globalCtx, CAM_ID_MAIN), 9);
+}
+
+s32 func_80123F14(GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+
+    return player->stateFlags1 & PLAYER_STATE1_800000;
+}
+
+s32 func_80123F2C(GlobalContext* globalCtx, s32 ammo) {
+    globalCtx->unk_1887C = ammo + 1;
+
+    return 1;
+}
+
+s32 Player_IsBurningStickInRange(GlobalContext* globalCtx, Vec3f* pos, f32 xzRange, f32 yRange) {
+    Player* this = GET_PLAYER(globalCtx);
+    Vec3f diff;
+    s32 pad;
+
+    if ((this->itemActionParam == PLAYER_AP_STICK) && (this->unk_B28 != 0)) {
+        Math_Vec3f_Diff(&this->meleeWeaponInfo[0].tip, pos, &diff);
+        return ((SQ(diff.x) + SQ(diff.z)) <= SQ(xzRange)) && (0.0f <= diff.y) && (diff.y <= yRange);
+    }
+
+    return false;
+}
+
+u8 Player_GetStrength(void) {
+    return D_801BFF98[(void)0, gSaveContext.playerForm];
+}
+
+u8 Player_GetMask(GlobalContext* globalCtx) {
```

Comment:
I suppose this should really be GetCurrentMask.

Reply:
I think `Player_GetMask` is fine, but I guess it could be changed.
I would like to know what does other people think about it first

Reply:
I'd agree. `Player_GetMask` is fine, context is enough.